### PR TITLE
Changing game year and making languages lore-friendly

### PR DESCRIPTION
**Pull request by Terror4000rus**
Removed standart BS12 languages from selection list and added some from language.dm in halo category.
English, Mandarin, French and German for now.
I can go further and remove sol common, gutter and other legacy languages, but I don't know devs reaction for now, so did a minimal work.
Oh, and also moved game year to 2540.

:cl: Terror4000rus
rscadd: Added new lore-friendly languages for humans, such as: english, germany, french, mandarin. Every but first one are optional.
rscdel: Removed Galactic Common, Independent and many other legacy BS12 languages from selection list. Let's pray it will not break everything.
bugfix: Fixed huragoks languages. They can understand everyone around, but cannot speak but at sign language.
maptweak: Replaced ODP silver walls with reinforced one (and painted).
spellcheck: Changed game year to 2540.
/:cl:

### CODE REVIEW

File: code/__defines/species_languages_halo.dm [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+#define LANGUAGE_INDONESIAN "Indonesian"
+#define LANGUAGE_SWEDISH    "Swedish"
+#define LANGUAGE_DANISH     "Danish"
+#define LANGUAGE_NORWEGIAN  "Norwegian"
+#define LANGUAGE_ICELANDIC  "Icelandig"
+#define LANGUAGE_FINNISH    "Finnish"
+#define LANGUAGE_JAPANESE   "Japanese"
+
+#define LANGUAGE_SANGHEILI  "Sangheili"
+#define LANGUAGE_LEKGOLO    "Lekgolo"
```
Comment (caelaislinn) [inline code/__defines/species_languages_halo.dm:17]: If you are putting Covenant languages here, there is a bunch of others including Unggoy, Sanshyuum, Brute etc

**Review by caelaislinn — CHANGES_REQUESTED** [review]
Looking good. This has been on my todo list after I looked at them about 1-2 years ago, but it's been low priority. The other B12 languages can go except for tradeband and gutter, which could be interesting background languages to add more flavour to the setting.

**Review by caelaislinn — CHANGES_REQUESTED** [review]
Changes needed to be Travis compatible: It looks like you gave English, Mandarin and German conflicting keys (1, 2 and 3) as those are already taken by galcom, tradeband and gutter. You'll need to move tradeband and gutter onto different non-conflicting keys and probably ditch galcom. 

As an idea for a future language related project, you could have a go at the Covenant language symbols I was experimenting with. You can see the remains of my attempt in `code/modules/halo/covenant/_language.dm`

**Comment by bloxgate** [comment]
As a suggestion on the Covenant language symbols, we have gained support for Unicode since that was attempted. One option may be to render it as one of the non-Latin alphabets we now have access to. While it might not be the most lore friendly, rendering the syllables as IPA symbols or something similar makes a bit more sense in my head than having people know the actual symbols that correspond to the sounds in Sangheili.

**Comment by Terror4000rus** [comment]
Updated

File: code/__defines/species_languages_halo.dm (hunk 1 above)
Comment (Terror4000rus) [reply code/__defines/species_languages_halo.dm:17]: Done.

**Review by bloxgate — DISMISSED** [review]
Automated changelogs should go into the PR description, in the format given in the PR template.

**Comment by Terror4000rus** [comment]
Updated

File: code/datums/trading/ai.dm [hunk 2]
```diff
@@ -9,7 +9,7 @@ They sell generic supplies and ask for generic supplies.
 /datum/trader/trading_beacon
 	name = "AI"
 	origin = "Trading Beacon"
-	name_language = LANGUAGE_EAL
+	name_language = LANGUAGE_ENGLISH
```
Comment (caelaislinn) [inline code/datums/trading/ai.dm:12]: These traders are unused by our code, but changing this is mostly harmless (except for making it a little harder to update from recent baycode)

**Review by caelaislinn — APPROVED** [review]

**Pull request merged**